cmds/core/gpgv/gpgv.go

```diff
@@ -113,10 +83,50 @@ func verifyDetachedSignature(key *packet.PublicKey, contentf, sigf io.Reader) er
 	switch sig := p.(type) {
 	case *packet.Signature:
 		err = key.VerifySignature(h, sig)
-	case *packet.SignatureV3:
-		err = key.VerifySignatureV3(h, sig)
 	default:
-		panic("unreachable")
+		return fmt.Errorf("unknown signature")
 	}
 	return err
 }
+
+func runGPGV(w io.Writer, verbose bool, keyfile, sigfile, datafile string) error {
+	if keyfile == "" || sigfile == "" || datafile == "" {
+		return errUsage
+	}
+
+	keyf, err := os.Open(keyfile)
+	if err != nil {
+		return err
+	}
+	defer keyf.Close()
+
+	sigf, err := os.Open(sigfile)
+	if err != nil {
+		return err
+	}
+	defer sigf.Close()
+
+	contentf, err := os.Open(datafile)
+	if err != nil {
+		return err
+	}
+	defer contentf.Close()
+
+	key, err := readPublicSigningKey(keyf)
+	if err != nil {
+		return fmt.Errorf("key: %v ", err)
+	}
+
+	if err = verifyDetachedSignature(key, contentf, sigf); err != nil {
+		return fmt.Errorf("verify: %v", err)
+	}
+	fmt.Fprintf(w, "OK")
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	if err := runGPGV(os.Stdout, verbose, flag.Args()[0], flag.Args()[1], flag.Args()[2]); err != nil {
```

Comment:
rminnich@a300:~/go/src/github.com/u-root/u-root/cmds/core/gpgv$ ./gpgv 
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.main()
	/home/rminnich/go/src/github.com/u-root/u-root/cmds/core/gpgv/gpgv.go:12
rminnich@a300:~/go/src/github.com/u-root/u-root/cmds/core/gpgv$